### PR TITLE
Docs/update

### PR DESCRIPTION
**Pull request by lucashervier**
Tackle issue #71. If we take point by point:

minor

# Some missing elements in the documentation

### Visibility of single method tutoriels

- [x] In the readme, there are tables referencing all methods, maybe adding a column with the link to the tutoriels will make them much more visible.

Done in 612373a 

### Common API documentation

- [x] A documentation on the attribution API, common for all attribution methods. As many thing are common (inheritance), a common documentation could be added. Detail on the fact that no common definition of attribution exist and that each method define one, thus some metrics tend to favorize some methods.
Done in 0b77a37
- [x] A documentation for the metrics API, same reason as above.
Done in f554e7a
- [ ] A documentation for the feature visualization API, same reason as above.
There is no API, everything already in the feature_viz.md
- [ ] A documentation for the concept based API, same reason as above.
Idem

### Points which need further explanations

- [x] In attribution method, the **targets** argument of the evaluate method is really important. Depending on its value, the aim of the explanation is completly different. This should be further explained.
Done in 6577d90 and 0b77a37
- [x] Some method or metrics can be far longer than others. It may be necessary to precise it, for example KernelShap or Stability.
Done in 6577d90 and 9fea21c

### Error or typo

- [x] The documentation of the __init__ method is not visible for Lime, KernelShap and Gradient Input.
Done in 6577d90

# In addition
## Fix minor css issues
Issue with table head in light mode, fix in 799b79d
## Fix previous docs error
A minor error in the Callable doc, fix in d3abfb9
## Update of mkdocs
Following the reorganization of the documentation, the navigation had to change a little bit see e371e67

### CODE REVIEW

File: docs/callable.md [hunk 1]
```diff
@@ -37,6 +37,7 @@ class TemplateTorchWrapper(nn.Module):
     return torch_inputs
 
 wrapped_model = TemplateTorchWrapper(torch_model)
+explainer = Lime(wrapped_model)
```
Comment (AntoninPoche) [inline docs/callable.md:40]: nice catch

**Review by AntoninPoche — COMMENTED** [review]
Great work !
- Great enhancement of the documentation, however, I do not know if I can see the results, I can only comment on the .md files.
- The improvments of the documentation wil enhance readibility and accessibility of the library, thank you !
- Great idea to add further details on methods arguments directly in the documentations.
- Error of pertubed and pertubation -> perturbed and perturbation (redoundant).
- I think that precising the targets shape in attribution methods documentation of the targets argument can be helpful.

File: xplique/attributions/base.py [hunk 2]
```diff
@@ -74,15 +74,26 @@ def explain(self,
...
+            documentation
+            (<a href=
+            "https://deel-ai.github.io/xplique/api/attributions/api_attributions/">
+            API Description </a>).
         targets
-            One-hot encoded labels or regression target (e.g {+1, -1}), one for each sample.
+            Tensor or Array. One-hot encoding of the model's output from which an explanation
+            is desired. One encoding per input and only one output at a time.
+            More information in the documentation
+            (<a href=
```
[the remark this answers is not in the document]
Comment (lucashervier) [reply xplique/attributions/base.py:94]: Done

File: xplique/attributions/kernel_shap.py [hunk 3]
```diff
@@ -18,59 +18,44 @@ class KernelShap(Lime):
...
+    nb_samples
+        The number of pertubed samples you want to generate for each input sample.
+        Default to 800.
+    ref_value
+        It defines reference value which replaces each feature when the corresponding
+        interpretable feature is set to 0.<br/>
+        It should be provided as: a ndarray of shape (1) if there is no channels in your input
+        and (C,) otherwise.<br/>
+        The default ref value is set to (0.5,0.5,0.5) for inputs with 3 channels (corresponding
+        to a grey pixel when inputs are normalized by 255) and to 0 otherwise.
```
[the remark this answers is not in the document]
Comment (lucashervier) [reply xplique/attributions/kernel_shap.py:50]: Thank you!

File: docs/callable.md (hunk 1 above)
Comment (lucashervier) [reply docs/callable.md:40]: Thank you!

File: docs/api/attributions/lime.md [hunk 4]
```diff
@@ -0,0 +1,173 @@
+# LIME
+
+!!!abstract "Tutorial"
+    <p style="text-align: center;">[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/drive/1InDzdW39-5k2ENfKqF2bs5qJEv8OJqi2?authuser=1)</p>
+
+The Lime method use an interpretable model to provide an explanation.
+More specifically, you map inputs ($x \in R^d$) to an interpretable space (e.g super-pixels) of size num_interpetable_features.
+From there you generate pertubed interpretable samples ($z' \in \{0,1\}^{num\_interpretable\_samples}$
```
[the remark this answers is not in the document]
Comment (lucashervier) [reply docs/api/attributions/lime.md:8]: Done

File: docs/api/attributions/kernel_shap.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+For instance you can use the scikit-image (as we did for the quickshift algorithm)
+library to defines super pixels on your images.
+
+!!!info
+    The quality of your explanation relies strongly on this mapping.
+
+!!!warning
+    Depending on the mapping you might have a huge number of `interpretable_features` 
+    (e.g you map pixels 2 by 2 on a 299x299 image). Thus, the compuation time might
+    be very long!
```
[the remark this answers is not in the document]
Comment (lucashervier) [reply docs/api/attributions/kernel_shap.md:81]: The case of tabular data is already mentionned since there is a default mapping function for that case.
I add the timeseries description though

**Review by AntoninPoche — APPROVED** [review]
LGTB ;)

File: docs/index.md [hunk 6]
```diff
@@ -138,6 +142,8 @@ concept_vector = extractor(positive_samples,
                            negative_samples)
 ```
 
+More information on CAV [here](https://deel-ai.github.io/xplique/api/concepts/cav.html) and on TCAV [here](https://deel-ai.github.io/xplique/api/concepts/tcav.html).
```
Comment (fel-thomas) [inline docs/index.md:145]: Super !

File: xplique/attributions/base.py [hunk 7]
```diff
@@ -74,15 +74,27 @@ def explain(self,
...
+            Dataset, Tensor or Array. Input samples to be explained.
+            If Dataset, targets should not be provided (included in Dataset).
+            Expected shape among (N, W), (N, T, W), (N, W, H, C). More information in the
+            documentation
+            (<a href=
+            "https://deel-ai.github.io/xplique/api/attributions/api_attributions/">
+            API Description </a>).
         targets
-            One-hot encoded labels or regression target (e.g {+1, -1}), one for each sample.
+            Tensor or Array. One-hot encoding of the model's output from which an explanation
```
Comment (fel-thomas) [inline xplique/attributions/base.py:91]: I found the doc easier before ^^ but I suppose I am biased on this one haha

File: xplique/attributions/base.py [hunk 8]
```diff
@@ -105,13 +117,16 @@ class WhiteBoxExplainer(BlackBoxExplainer, ABC):
...
-        Model used for computing explanations.
+        The model from which we want to obtain explanations
     output_layer
-        Layer to target for the output (e.g logits or after softmax), if int, will be be interpreted
-        as layer index, if string will look for the layer name. Default to the last layer, it is
-        recommended to use the layer before Softmax.
+        Layer to target for the outputs (e.g logits or after softmax).
+        If an `int` is provided it will be interpreted as a layer index.
+        If a `string` is provided it will look for the layer name.
+        <br/>
```
Comment (fel-thomas) [inline xplique/attributions/base.py:125]: Comments are made to explain, not to put more code. Is the <br> on purpose ?

**Review by fel-thomas — CHANGES_REQUESTED** [review]
Many changes, the documentation is greatly improved, bravo Lucas! 🎉 

For me, there are two things to modify and go merge: 
- remove the code in the comments (link to the doc or line break)
- the tutorial card with only the colab button in the middle, I propose an alternative in the comments. But at the UX level, a card with just a button in the middle is not something usual.

File: xplique/attributions/base.py (hunk 8 above)
Comment (fel-thomas) [reply xplique/attributions/base.py:125]: For my part, I am really against the idea of putting links to the doc in the code

Comment (lucashervier) [reply xplique/attributions/base.py:125]: It is just prettier when the actual doc is generated on the website and make it more readable. It understands HTML syntax but not markdown (you can try if you want). If it is just for the <br> I would not change it (but other complicated HTML syntax yes). But if in your opinion it is really important to not have it I can delete that!

File: xplique/attributions/base.py (hunk 7 above)
Comment (lucashervier) [reply xplique/attributions/base.py:91]: @AntoninPoche your opinion?

File: xplique/attributions/base.py [hunk 9]
```diff
@@ -74,15 +74,27 @@ def explain(self,
...
         Parameters
         ----------
         inputs
-            Input samples to be explained.
+            Dataset, Tensor or Array. Input samples to be explained.
+            If Dataset, targets should not be provided (included in Dataset).
+            Expected shape among (N, W), (N, T, W), (N, W, H, C). More information in the
+            documentation
+            (<a href=
+            "https://deel-ai.github.io/xplique/api/attributions/api_attributions/">
```
[the remark this answers is not in the document]
Comment (lucashervier) [reply xplique/attributions/base.py:88]: OK, I can delete it

File: xplique/attributions/base.py (hunk 7 above)
Comment (AntoninPoche) [reply xplique/attributions/base.py:91]: I think the most important thing is that the user does not mix targets with the ground truth or the model predictions. I think that a line about this may be needed.
Nevertheless, I prefer Lucas' doc, I think that {-1, 1} is misleading as without further information, -1 should not be put in the targets. (It is not trivial to interpret).

Comment (fel-thomas) [reply xplique/attributions/base.py:91]: Ok go for the new one then ;)

File: docs/api/attributions/grad_cam.md [hunk 10]
```diff
@@ -1,5 +1,8 @@
 # Grad-CAM
 
+!!!abstract "Tutorial"
+    <p style="text-align: center;">[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/drive/1wJmPK3HZ1Uw1rBbeo8DtmmaRZUZ1Cs-i?authuser=1)</p>
+
```
[the remark this answers is not in the document]
Comment (lucashervier) [reply docs/api/attributions/grad_cam.md:5]: Nice catch!

File: docs/api/attributions/grad_cam.md [hunk 11]
```diff
@@ -1,5 +1,8 @@
 # Grad-CAM
 
+!!!abstract "Tutorial"
```
[the remark this answers is not in the document]
Comment (lucashervier) [reply docs/api/attributions/grad_cam.md:3]: Done!

File: README.md [hunk 12]
```diff
@@ -107,6 +107,8 @@ explanations = explainer.explain(images, labels)
 # or just `explainer(images, labels)`
 ```
 
+All attributions methods share a common API. You can find out more about it [here](https://deel-ai.github.io/xplique/api/attributions/api_attribution.html).
```
[the remark this answers is not in the document]
Comment (lucashervier) [reply README.md:110]: Well it is on a separate page :) I marked it as resolved except if you wanted something very particular (in which case I did not really understand ^^)

File: docs/api/metrics/api_metrics.md [hunk 13]
```diff
@@ -0,0 +1,74 @@
+# API: Metrics
+
```
[the remark this answers is not in the document]
Comment (lucashervier) [reply docs/api/metrics/api_metrics.md:2]: Done

**Comment by lucashervier** [comment]
Change required are done!

**Review by fel-thomas — APPROVED** [review]
